### PR TITLE
[docs-beta] Merge the new beta Docs site

### PR DESCRIPTION
**Pull request by PedramNavid**
## Summary & Motivation

This adds the new beta docs site into our main repo. This is still a work-in-progress, and does not affect our existing docs deployment. This work was originally undertook in the #23530 branch, however, I have brought it in as a squashed self-contained PR to keep the history clean.

## How I Tested These Changes

Built locally, BK, and Vercel.

### CODE REVIEW

File: .github/workflows/vale.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  vale:
+    name: runner / vale
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: errata-ai/vale-action@reviewdog
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          files: '["docs-next/docs"]'
```
Comment (cmpadden) [inline .github/workflows/vale.yml:32]:
```suggestion
          files: '["docs-beta/docs"]'
```

File: docs/docs-beta/static/img/docusaurus-social-card.jpg
Comment (cmpadden) [inline docs/docs-beta/static/img/docusaurus-social-card.jpg:1]: can remove

File: docs/docs-beta/static/img/docusaurus.png
Comment (cmpadden) [inline docs/docs-beta/static/img/docusaurus.png:1]: can remove - though very cute

**Review by cmpadden — APPROVED** [review]

File: docs/docs-beta/static/img/docusaurus-social-card.jpg
Comment (PedramNavid) [reply docs/docs-beta/static/img/docusaurus-social-card.jpg:1]: good call

**Pull request merged**